### PR TITLE
Add support for irods:: style AVUs

### PR DESCRIPTION
**Pull request by kjsanger**
These are added by iRODS' own internal processes and are exposed amongst user AVUs.

### CODE REVIEW

File: src/partisan/irods.py [hunk 1]
```diff
@@ -841,14 +849,33 @@ def __init__(self, attribute: Any, value: Any, units=None, namespace=None):
         if namespace is None:
             namespace = ""
 
-        if namespace.find(AVU.SEPARATOR) >= 0:
+        attr = str(attribute)
+        value = str(value)
+
+        if re.match(r"\s+$", attr):
+            raise ValueError("AVU attribute may not be entirely whitespace")
```
Comment (marcomoscasgr) [inline src/partisan/irods.py:856]: I couldn't find whether they are already covered, do you also need to add tests for this type of strings with whitespaces?

File: src/partisan/irods.py [hunk 2]
```diff
@@ -841,14 +849,33 @@ def __init__(self, attribute: Any, value: Any, units=None, namespace=None):
...
+        # Handle all other AVUs, namespaced or not
+        else:
+            self._separator = AVU.SEPARATOR
+
+        if namespace and namespace.find(self._separator) >= 0:
             raise ValueError(
-                f"AVU namespace contained a separator '{AVU.SEPARATOR}': '{namespace}'"
+                f"AVU namespace contained a separator '{self._separator}': "
+                f"'{namespace}'"
             )
```
Comment (marcomoscasgr) [inline src/partisan/irods.py:875]: do you need to add a test for this too?

File: src/partisan/irods.py [hunk 3]
```diff
@@ -944,7 +971,7 @@ def attribute(self):
         """The attribute, including namespace, if any. The namespace an attribute are
         separated by AVU.SEPARATOR."""
         if self._namespace:
-            return f"{self._namespace}{AVU.SEPARATOR}{self._attribute}"
+            return f"{self._namespace}{self._separator}{self._attribute}"
```
Comment (marcomoscasgr) [inline src/partisan/irods.py:974]: Do you need to change the function description after this update?

File: src/partisan/irods.py (hunk 2 above)
Comment (kjsanger) [reply src/partisan/irods.py:875]: There is a test for this already

**Review by marcomoscasgr — APPROVED** [review]

**Pull request merged**